src/inventory.rs

```diff
@@ -0,0 +1,197 @@
+use std::ops::Range;
+
+use crate::slot::{Slot, SlotId};
+
+pub trait Inventory {
+    fn get_slot(&self, slot_id: SlotId) -> &Slot;
+    fn set_slot(&mut self, slot_id: SlotId, slot: Slot);
+    fn capacity(&self) -> usize;
+
+    // TODO: `entry()` style api
+}
+
+pub trait CraftingInventory {
+    fn craft_result_slot() -> SlotId;
+    fn craft_table_slots() -> Range<SlotId>;
+}
+
+/// Represents a player's Inventory.
+#[derive(Debug, Clone)]
+pub struct PlayerInventory {
+    pub(crate) slots: [Slot; 46],
```

Comment:
If `PlayerInventory` is allowed to be owned by library users (Which is the case currently because of the `Default` impl) then we should probably box this field. Don't want to be passing this massive amount of data around on the stack.
```suggestion
    pub(crate) slots: Box<[Slot; 46]>,
```